static/app/views/onboarding/setupDocs.tsx

```diff
@@ -451,3 +562,47 @@ const MainContent = styled('div')`
   min-width: 0;
   flex-grow: 1;
 `;
+
+const SyntaxHighlight = styled(CodeSnippet)`
+  margin-top: 1em;
+  margin-bottom: 1em;
+`;
+
+const GettingStartedDocWrapper = styled('div')<{activeProductSelection: PRODUCT[]}>`
+  hr {
+    border-color: ${p => p.theme.border};
+  }
+
+  .replay,
+  .performance,
+  .replay-and-performance,
+  .error-monitoring {
+    display: none;
+  }
+
+  ${p =>
+    p.activeProductSelection.includes(PRODUCT.SESSION_REPLAY) &&
+    p.activeProductSelection.includes(PRODUCT.PERFORMANCE_MONITORING)
+      ? css`
+          .replay-and-performance {
+            display: block;
+          }
+        `
+      : p.activeProductSelection.includes(PRODUCT.SESSION_REPLAY)
+      ? css`
+          .replay {
+            display: block;
+          }
+        `
+      : p.activeProductSelection.includes(PRODUCT.PERFORMANCE_MONITORING)
+      ? css`
+          .performance {
+            display: block;
+          }
+        `
+      : css`
+          .error-monitoring {
+            display: block;
+          }
+        `}
```

Comment:
The mdx is nice, but this is the weird part imo. 

I don't love how it's separate from the mdx file, and i'm not sure how we could extend it to support more frameworks and languages, with different next-steps and possible name collisions.

Would it work if we also include `activeProductSelection` when we render the mdx. Maybe we need a helper react component to do the show/hide or maybe bare css in the `<div>` would work
```diff
        <GettingStartedDoc
          dsn={projectKeys[0].dsn.public}
+         activeProductSelection={products}
          components={{
            code: MDXCodeSyntaxHighlight,
            pre: MDXPre,
          }}
        />
```

Reply:
Yes, I completely agree with your point, and one alternative I can suggest is to dynamically load separate code in the .mdx file based on the selected product. In other words, we can dynamically load portions of .mdx files into the main .mdx file. Does that idea make sense ? In this way we don't need to use `css` at all

Reply:
> Would it work if we also include activeProductSelection when we render the mdx. Maybe we need a helper react component to do the show/hide or maybe bare css in the <div> would work

We could theoretically use `if...else` but this would turn things more challenge for non-developers to maintain the doc

Reply:
Dynamically loading would be great! I was also thinking, since it's mdx, the helper component could literally be the if-statement itself:
```
function Section({children, activeProductSelection, name}) {
  return activeProductSelection.includes(name)
    ? children
    : null;
}
```
and hopefully the mdx is:
```diff
  ## Configure

  Initialize Sentry as early as possible in your application's lifecycle.

- <div className='error-monitoring'>
+ <Section name="error-monitoring" activeProductSelection={activeProductSelection}> 

```javascript
import { createRoot } React from "react-dom/client";
import React from "react";
... etc ...

- </div>
+ </Section>
```

Either wait, this brings up a good point that we should ask the dev-rel team and docs team for feedback on all the different approaches to the content files

Reply:
I like your suggestion! will apply. Thank you!!